File: docs/validation/go/app/scenarios/av04_missing_group.go
package scenarios

// AV-04: Reader returns an error when required group (Machine/) is absent

import (
	"fmt"

	mc "machine-config-go"
)

func RunAv04MissingGroup(fixturesDir, _ string) (bool, string) {
	fixture := avFixture(fixturesDir, "missing_machine_group.h5")
	_, err := mc.NewReader(fixture).Parse()
	if err == nil {
		return false, "no error raised for missing Machine/ group"
	}
	return true, fmt.Sprintf("error raised for missing Machine/ group: %v", err)
}

File: docs/validation/rust/app/src/scenarios/av04_missing_group.rs
// AV-04: Reader returns typed error when required group (Machine/) is absent
use super::common::av_fixture;
use machine_config::MachineConfigReader;
use std::path::Path;

pub fn run(fixtures_dir: &Path, _real_dir: &Path) -> (bool, String) {
    let fixture = av_fixture(fixtures_dir, "missing_machine_group.h5");
    match MachineConfigReader::open(&fixture).and_then(|r| r.parse()) {
        Ok(_) => (false, "no error raised for missing Machine/ group".to_string()),
        Err(e) => (true, format!("error raised for missing Machine/ group: {e}")),
    }
}
